docs/tutorial/overview.md

```diff
@@ -28,33 +27,112 @@ component main() -> {
 - `incr_idx`: Increments an `idx` register using an adder. This group is done when the `idx` register is written to.
-- `cond`: Applies a "less than" operator to `idx`, and the length of our input arrays.
+- `cond`: Applies a "less than" operator to `idx`, and the length of our input arrays, using the `le` hardware unit.
 - `eval_body`: Reads from an array, performs some kind of computation, and writes the result of the computation to an accumulator register or another array.
 
-We'll make these groups for each `Map` and `Reduce` node, so to avoid naming collisions, we'll suffix each group with an integer starting at 0, incrementing each time we need to add new sets of  groups. These groups will be added to the `wires` section. We'll also need to add logic to the `control` section as well that uses these groups to process arrays:
+We'll make these groups for each `Map` and `Reduce` node, so to avoid naming collisions, we'll suffix each group with an integer starting at 0, incrementing each time we need to add a new set of  groups. These groups will be added to the `wires` section. We'll also need to add logic to the `control` section as well that uses these groups to process arrays:
+
 ```
 while le0.out with cond0 {
   seq { eval_body0; incr_idx0; }
 }
 ```
 
+### Parallelization
+
+MrXL allows you to parallelize your `map` and `reduce` operations. Let's revisit the `map` example from earlier:
+
+```
+input foo: int[4]
+output baz: int[4]
+baz := map 4 (a <- foo) { a + 5 }
+```
+
+The number 4 after the `map` specifies the number of adders we can use at once to parallelize this computation. There are a few ways we could parallelize this program, and one of them is to split the memories used in the `map` operation into 4 separate memory _banks_, and then we can read from each bank of `foo` and write into each bank of `baz` simultaneously. In general, we can break memories of size `m` into `b` banks (each with size `m/b`), and then simultaneously process those `b` banks. Realizing this in FuTIL means creating separate memories for each bank, and creating `group`s to process each bank. In the `Map` and `Reduce` code generation section we described `group`s that could be orchestrated to iterate over a memory and process it. We'll now have to do that for each memory bank, and then parallelize these operations in the generated FuTIL's `control` section:
+
+```
+par {
+  while le_b0.out with cond_b0 { seq { eval_body_b0; incr_idx_b0; } }
+  while le_b1.out with cond_b1 { seq { eval_body_b1; incr_idx_b1; } }
+  while le_b2.out with cond_b2 { seq { eval_body_b2; incr_idx_b2; } }
+  while le_b3.out with cond_b3 { seq { eval_body_b3; incr_idx_b3; } }
+}
+```
 
+### Scheduling
```

Comment:
What is scheduling supposed to refer to? Without HLS background, I don't see any relationship between complex expressions and scheduling.